src/lib/utils.js

```diff
@@ -10,3 +10,8 @@ export const colorToCSS = color => {
     ? `hsl(${h}, ${s}%, ${l}%)`
     : `hsla(${h}, ${s}%, ${l}%, ${color.a * 0.01})`;
 };
+
+export const themesEqual = (themeA, themeB) =>
```

Comment:
Not using this yet, which reminds me of the other thing I'm trying to figure out: When to decide that the shared theme matches the current theme.

I tried doing it as kind of a validation step - the pendingTheme state would not accept a value that matches the current theme. But that ran into race conditions in add-on messaging. Could just check on an arbitrary delay and auto-dismiss the accept dialog, but then there's a flash of content

Reply:
This seems like a good usecase for a selector; just have a boolean prop passed to the `<App>` that checks the `pendingTheme` and `currentTheme` for equality. Then it's just a matter of checking that prop, as passed through the component hierarchy.

Reply:
I also tried this as a selector. Problem was, as soon as you change any colors, suddenly the pending theme no longer matches the current theme and the approval dialog appears. Been trying to figure out at what point to just ignore / delete the incoming pendingTheme so that doesn't happen